processing/src/main/java/io/druid/segment/incremental/TimeAndDimsHolder.java

```diff
@@ -17,19 +17,33 @@
  * under the License.
  */
 
-package io.druid.segment.historical;
+package io.druid.segment.incremental;
 
-import io.druid.segment.data.Offset;
-import io.druid.segment.data.ReadableOffset;
-
-public interface OffsetHolder
+/**
+ * This interface is the core "pointer" interface that is used to create {@link io.druid.segment.ColumnValueSelector}s
+ * over incremental index. It's counterpart for historical segments is {@link io.druid.segment.data.Offset}.
+ */
+public class TimeAndDimsHolder
 {
-  Offset getOffset();
+  IncrementalIndex.TimeAndDims currEntry = null;
+
+  public IncrementalIndex.TimeAndDims get()
+  {
+    return currEntry;
+  }
+
+  public void set(IncrementalIndex.TimeAndDims currEntry)
+  {
+    this.currEntry = currEntry;
+  }
+
+  public IncrementalIndex.TimeAndDims getKey()
+  {
+    return currEntry;
```

Comment:
Is this method supposed to return ```currEntry``` as ```get()``` does? Also, what do the key and value mean here?

Reply:
I agree that this class has strange API. In the future PRs in this series I will probably reduce it to just get() + set(). However in this PR I just renamed and moved this class to upper level, without other modifications.

Reply:
Ok, then please leave a comment about this.

Reply:
Added comments